service_grpc_influxrpc/src/service.rs

```diff
@@ -1586,6 +1595,58 @@ where
     }
 }
 
+/// Chunk given [`ReadResponse`]s -- while preserving the [`Frame`] order -- into responses that shall at max have the
+/// given size.
+///
+/// # Panic
+/// Panics if `size_limit` is 0.
+fn chunk_read_responses(responses: Vec<ReadResponse>, size_limit: usize) -> Vec<ReadResponse> {
+    assert!(size_limit > 0);
+
+    let mut out = Vec::with_capacity(1);
+    let it = responses
+        .into_iter()
+        .flat_map(|response| response.frames.into_iter());
+
+    let mut frames = vec![];
+    let mut size = 0;
+    for frame in it {
+        let fsize = frame_size(&frame);
+
+        // flush?
+        if size + fsize > size_limit {
+            let mut tmp = vec![];
+            std::mem::swap(&mut tmp, &mut frames);
```

Comment:
I think you can write this pattern using `take`, which may be slightly easier to read

```suggestion
            let frames =  std::mem::take(&mut frames);
```

Reply:
done